dev_hero_selector/gui.py

```diff
@@ -0,0 +1,140 @@
+)
+
+
+class PlayerPanel(wx.Panel):
+    def __init__(self, parent: 'ScoreboardGui', data: Player) -> None:
+        wx.Panel.__init__(self, parent)
+        self.parent = parent
+        self.data = data
+
+        name = wx.TextCtrl(self, value=data['name'])
+        dropdown = wx.ComboBox(self, value=data['current_hero'], choices=heroes, style=wx.CB_READONLY)
+        kills_label = wx.StaticText(self, label="Kills")
+        kills = wx.SpinCtrl(self)
+        deaths_label = wx.StaticText(self, label="Deaths")
+        deaths = wx.SpinCtrl(self)
+        res_label = wx.StaticText(self, label="Resurrects")
+        res = wx.SpinCtrl(self)
+        res.Disable()
+
+        sizer = wx.BoxSizer(wx.VERTICAL)
+        sizer.Add(name, 0, wx.EXPAND | wx.ALL, 5)
+        sizer.Add(dropdown, 0, wx.EXPAND | wx.ALL, 5)
+
+        grid_sizer = wx.FlexGridSizer(3, 2, 0, 0)
+        grid_sizer.Add(kills_label, 0, wx.EXPAND | wx.ALL, 5)
+        grid_sizer.Add(kills, 0, wx.EXPAND | wx.ALL, 5)
+        grid_sizer.Add(deaths_label, 0, wx.EXPAND | wx.ALL, 5)
+        grid_sizer.Add(deaths, 0, wx.EXPAND | wx.ALL, 5)
+        grid_sizer.Add(res_label, 0, wx.EXPAND | wx.ALL, 5)
+        grid_sizer.Add(res, 0, wx.EXPAND | wx.ALL, 5)
+        sizer.Add(grid_sizer)
+
+        self.SetSizerAndFit(sizer)
+        self.res = res
+
+        name.Bind(wx.EVT_TEXT, self.set_name)
+        dropdown.Bind(wx.EVT_COMBOBOX, self.set_hero)
+        kills.Bind(wx.EVT_SPINCTRL, self.set_kills)
+        deaths.Bind(wx.EVT_SPINCTRL, self.set_deaths)
+        res.Bind(wx.EVT_SPINCTRL, self.set_res)
+
+    def set_name(self, event: wx.CommandEvent) -> None:
+        self.data['name'] = event.GetString()
+        self.parent.update()
+
+    def set_hero(self, event: wx.CommandEvent) -> None:
+        hero = event.GetString()
+        self.data['current_hero'] = hero
+        if hero == 'mercy':
+            self.res.Enable()
+            self.data['resurrects'] = self.res.GetValue()
+        else:
+            self.res.Disable()
+            self.data['resurrects'] = None
+        self.parent.update()
+
+    def set_kills(self, event: wx.SpinEvent) -> None:
+        self.data['kills'] = event.GetPosition()
+        self.parent.update()
+
+    def set_deaths(self, event: wx.SpinEvent) -> None:
+        self.data['deaths'] = event.GetPosition()
+        self.parent.update()
+
+    def set_res(self, event: wx.SpinEvent) -> None:
+        self.data['resurrects'] = event.GetPosition()
+        self.parent.update()
+
+
+class ScoreboardGui(wx.Frame):
+    def __init__(self, *, callback: Optional[Callback] = None) -> None:
+        wx.Frame.__init__(self, None)
+        self.callback = callback
+        self._data: List[Player] = []
+
+        sizer = wx.GridSizer(2, 6, 0, 0)
+
+        for team in range(2):
+            for player in range(6):
+                data = {
+                    'name': f'{["Blue", "Red"][team]} {player + 1}',
+                    'current_hero': 'dva',
```

Comment:
Use a different hero for each player by default - perhaps `heroes[team*6+i]`.
Might also be good to force include mercy for the res so people are less likely to break that.